### ISSUE TITLE
[Iceberg] Always "CannotDeleteFileException" when trying to remove a repository and these files 

### ISSUE THREAD

**Issue by labordep** [guillep/pharo-core#15990]
![image](https://github.com/pharo-project/pharo/assets/49183340/bbf246cf-2edd-4c37-948a-d492d5dc0642)

My use-case in the video below: I load a repository using a script, and finally I want to remove it.

1.  Load a repository using a playground, for example: 

```smalltalk
Metacello new
	repository: 'github://juliendelplanque/SQLite3/src';
	baseline: 'SQLite3';
	load
```

2. (optionnal) Unload each packages to remove the code
3. Remove the repository from Iceberg, I want to remove file from the disk, and Error!
![image](https://github.com/pharo-project/pharo/assets/49183340/838c3692-f406-44a4-bc87-8ad2d9788388)

This problem keeps happening **every time** I try to delete files from the repository.

Video with the example:
https://github.com/pharo-project/pharo/assets/49183340/c458b549-e41b-4089-aa43-e7e933304637

My config:
- Pharo-11.0.0+build.721
- VM 10.0.9
- Windows 11 Pro 64bits

**Comment by Ducasse** [comment]
I do not think that this is related to iceberg but file or FileReference. Did you succeed to delete the file when you use File deleteFile:? 
What are the file attributes? Because we cannot debug it since for example on my machine this is always working.

**Comment by labordep** [comment]
You are likely correct because when attempting to delete various items, such as images or repositories, on Windows, we encounter this error. This is probably attributed to the file management approach used for Windows.
I am surprised that no one has raised this issue yet. We have been dealing with it for years, but I would really like to help fixing that because it's really cumbersome.

**Comment by labordep** [comment]
I'm testing that: 

```smalltalk
File deleteFile: 'C:\Users\labor\Documents\Pharo\images\SQLite3\pharo-local\iceberg\pharo-rdbms\Pharo-SQLite3\src\SQLite3-Core'
```

And I have the exception :)

![image](https://github.com/pharo-project/pharo/assets/49183340/aa16469c-e6f5-4ef9-b3cc-ee50ece1c3b3)

**Comment by labordep** [comment]
But it is probably normal to have this error at this point, maybe Iceberg place locks on files during Pharo execution ? I don't know.

**Comment by Ducasse** [comment]
I do it often and I do not get the problems so there is something more. May be the libgit it doing something special.

**Comment by Ducasse** [comment]
We should try on windows.

**Comment by LabordePierre** [comment]
@Ducasse this is related to the Windows file management problem as I explained to you :)
May be @ELePors have another example to help to reproduce.

**Comment by Ducasse** [comment]
This is a good example where we can improve the windows API and file support.

**Comment by ELePors** [comment]
If the iceberg load and delete repository are done when no restart of Pharo is done, the git library keep open a file. When everything is finished with git, Pharo should release git library. 
I reproduce each time i load a git repo... and just after, because i finally dont want the code in my image, i finish to remove the code and the repo folder... -> bug.

Cheers,
Eric.

**Comment by ELePors** [comment]
@guillep @tesonep @Ducasse 
after a git clone, if you don't close your Pharo image, a file handle is still opened on a .pack... file in the iceberg .git folder of the repository. Then if you try to remove it, pharo says : CannotDeleteFileException
I think the lock is made on this file by git. The solution should be to release correctly git files after the clone ... The other solution we have is to close the Pharo image and to reopen it, the file is not locked anymore... 
but ... wow ! :)

I think it is inside a git command (library call ... ) not directly in pharo... 

Cheers,
Eric.
